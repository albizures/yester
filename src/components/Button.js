import React from 'react'
import PropTypes from 'prop-types'
import { View, TouchableHighlight, Image, ViewPropTypes, StyleSheet } from 'react-native'
import colors from '../utils/colors'
import { Heading3 } from '../components'

export const types = {
  FILLED: 'FILLED',
  OUTLINED: 'OUTLINED',
}

export const sizes = {
  NORMAL: 'NORMAL',
  SMALL: 'SMALL',
}

const states = {
  ENABLED: 'ENABLED',
  DISABLED: 'DISABLED',
}

const Button = (props) => {
  const { title, type, size, icon, disabled } = props
  const typeStyle = typeStyles[disabled ? states.DISABLED : states.ENABLED][type]
  const sizeStyle = sizeStyles[size]

  return (
    <TouchableHighlight
      {...props}
      style={[styles.button, typeStyle.button, sizeStyle.button, props.style]}
    >
      <View style={styles.row}>
        {icon && <Image source={icon} style={[styles.icon, typeStyle.icon]} />}
        <Heading3 keyName={title} style={[styles.text, typeStyle.text]} />
      </View>
    </TouchableHighlight>
  )
}

Object.assign(Button, types, sizes, states)

Button.propTypes = {
  style: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(Object.keys(types)),
  icon: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  size: PropTypes.oneOf(Object.keys(sizes)),
  disabled: PropTypes.bool,
}

Button.defaultProps = {
  type: types.FILLED,
  size: sizes.NORMAL,
  disabled: false,
}

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    height: 50,
    borderRadius: 24,
    borderWidth: 2,
    justifyContent: 'center',
    shadowColor: colors.brightTurquoise,
    shadowOpacity: 0.22,
    shadowRadius: 15,
    shadowOffset: {
      height: 10,
    },
    elevation: 10,
  },
  text: {
    textAlign: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    marginRight: 9,
  },
})

const typeStyles = {
  [states.ENABLED]: {
    [types.FILLED]: StyleSheet.create({
      button: {
        backgroundColor: colors.governorBay,
        borderColor: colors.governorBay,
      },
      text: {
        color: colors.white,
      },
      icon: {
        tintColor: colors.white,
      },
    }),
    [types.OUTLINED]: StyleSheet.create({
      button: {
        backgroundColor: colors.white,
        borderColor: colors.governorBay,
      },
      text: {
        color: colors.governorBay,
      },
      icon: {
        tintColor: colors.governorBay,
      },
    }),
  },
  [states.DISABLED]: {
    [types.FILLED]: StyleSheet.create({
      button: {
        backgroundColor: colors.mischka,
        borderColor: colors.mischka,
      },
      text: {
        color: colors.white,
      },
      icon: {
        tintColor: colors.white,
      },
    }),
    [types.OUTLINED]: StyleSheet.create({
      button: {
        backgroundColor: colors.white,
        borderColor: colors.mischka,
      },
      text: {
        color: colors.mischka,
      },
      icon: {
        tintColor: colors.mischka,
      },
    }),
  },
}

const sizeStyles = {
  [sizes.NORMAL]: StyleSheet.create({
    button: {
      width: 300,
    },
  }),
  [sizes.SMALL]: StyleSheet.create({
    button: {
      width: 150,
    },
  }),
}

export default Button
