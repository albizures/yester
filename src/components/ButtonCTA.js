import React from 'react'
import PropTypes from 'prop-types'
import { TouchableHighlight, ViewPropTypes, Text } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

const ButtonCTA = (props) => {
  return (
    <TouchableHighlight {...props} elevation={10} style={[props.style, styles.button]}>
      <LinearGradient start={{x: 0.0, y: 0.0}} end={{x: 1.0, y: 1.0}} style={styles.gradient} colors={['#c79a63', '#d9b992']}>
        <Text style={styles.text}>{props.title}</Text>
      </LinearGradient>
    </TouchableHighlight>
  )
}

ButtonCTA.propTypes = {
  style: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
}

const styles = {
  gradient: {
    flex: 1,
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 30,
  },
  button: {
    height: 50,
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    borderRadius: 30,
    backgroundColor: '#b98a56',
    width: '100%',
  },
  text: {
    // marginVertical: 20,
    color: '#252525',
    fontWeight: 'bold',
    fontSize: 16,
  },
}

export default ButtonCTA
