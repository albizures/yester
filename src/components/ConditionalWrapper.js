import React from 'react';
import PropTypes from 'prop-types';

const ConditionalWrapper = ({ component, props, condition, children }) => {
	const Component = component;
	return condition ? <Component {...props}>{children}</Component> : children;
};

ConditionalWrapper.propTypes = {
	component: PropTypes.func.isRequired,
	condition: PropTypes.bool,
	children: PropTypes.node.isRequired,
	props: PropTypes.object,
};

export default ConditionalWrapper;
