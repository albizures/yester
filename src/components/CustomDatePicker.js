import React from 'react'
import PropTypes from 'prop-types'
import { View, ViewPropTypes, StyleSheet, Image, TouchableOpacity } from 'react-native'
import colors from '../utils/colors'
import Translate from '../components/Translate'
import icons from '../utils/icons'
import DatePicker from 'react-native-datepicker'

const states = {
  ENABLED: 'ENABLED',
  DISABLED: 'DISABLED',
}

export const validations = {
  NONE: 'NONE',
  GOOD: 'GOOD',
  WAIT: 'WAIT',
  BAD: 'BAD',
}

const CustomDatePicker = (props) => {
  const {title, editable, description, value, validation,
    validationMessage, show} = props
  const stateStyle = stateStyles[editable ? states.ENABLED : states.DISABLED]
  const validationStyle = validationStyles[validation]

  var descriptionLabel
  if (description != null) {
    descriptionLabel = <Translate keyName={description}
      style={styles.description} />
  }

  var validationLabel
  if (validation !== validations.NONE) {
    validationLabel = <Translate keyName={validationMessage}
      style={[styles.validation, validationStyle.validation]} />
  }

  var showButton
  var onPress = () => {}
  if (show != null) {
    onPress = show
    showButton = <TouchableOpacity onPress={onPress}>
      <View style={[{flex: 1, width: 55, justifyContent: 'center'}]}>
        <Translate keyName='signup.showPassword' style={[{fontFamily: 'Karla-Regular',
          color: colors.royalBlue,
          fontSize: 14,
          fontWeight: 'bold',
          textAlign: 'left',
        }]} />
      </View>
    </TouchableOpacity>
  }

  return (
    <View style={[{marginVertical: 10}]} >
      <View style={[styles.labelContainer]}>
        <Translate keyName={title}
          style={[styles.tile, stateStyle.title, validationStyle.title]} />
      </View>
      <View style={[styles.input, stateStyle.input, validationStyle.input]}>
        <DatePicker {...props}
          date={value}
          mode='date'
          placeholder='YYYY-MM-DD'
          format='YYYY-MM-DD'
          showIcon={false}
          confirmBtnText='Confirm'
          cancelBtnText='Cancel'
          customStyles={{
            dateTouchBody: {
              flexDirection: 'row',
              height: 50,
              width: 300,
              alignItems: 'center',
              justifyContent: 'center',
            },
            dateIcon: {
              width: 32,
              height: 32,
              marginLeft: 5,
              marginRight: 5,
            },
            dateInput: {
              flex: 1,
              height: 50,
              width: 300,
              borderWidth: 0,
              borderColor: '#fff',
              alignItems: 'flex-start',
              justifyContent: 'center',
              paddingLeft: 20,
            },
            dateText: {
              fontSize: 18,
              color: colors.mineShaft,
            },
            placeholderText: {
              fontSize: 18,
              color: colors.mineShaft,
            },
          }}
        />
        {showButton}
        {// <Image source={icons.fb} style={{ width: 20, height: 20, marginRight: 9 }} />
        }
      </View>
      <View style={[{justifyContent: 'flex-end'}]}>{/* styles.labelContainer, */}
        {validationLabel}
      </View>
      <View style={[{justifyContent: 'flex-end'}]}>{/* styles.labelContainer, */}
        {descriptionLabel}
      </View>
    </View>
  )
}

CustomDatePicker.propTypes = {
  style: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
  editable: PropTypes.bool,
  description: PropTypes.string,
  value: PropTypes.string,
  validation: PropTypes.oneOf(Object.keys(validations)),
  validationMessage: PropTypes.string,
}

CustomDatePicker.defaultProps = {
  editable: true,
  validation: validations.NONE,
}

const styles = StyleSheet.create({
  input: {
    flex: 1,
    flexDirection: 'row',
    width: 300,
    maxHeight: 50,
    borderRadius: 25,
    shadowColor: colors.mischka,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowRadius: 15,
    shadowOpacity: 1.0,
    elevation: 10,
  },
  title: {
    position: 'absolute',
    left: 0,
    top: 0,
    fontSize: 12,
  },
  description: {
    fontSize: 12,
    color: colors.mineShaft,
  },
  labelContainer: {
    minHeight: 24,
  },
  validation: {
  },
})

const stateStyles = {
  [states.ENABLED]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.white,
      borderWidth: 0.5,
    },
    title: {
      color: colors.governorBay,
    },
    description: {
    },
    validation: {
    },
  },
  [states.DISABLED]: {
    input: {
      backgroundColor: colors.mischka,
      borderColor: colors.mischka,
      borderWidth: 0.5,
    },
    title: {
      color: colors.mischka,
    },
    description: {
    },
    validation: {
    },
  },
}

const validationStyles = {
  [validations.NONE]: {
    input: {
    },
    title: {
    },
    description: {
    },
    validation: {
    },
  },
  [validations.GOOD]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.apple,
      borderWidth: 2,
    },
    title: {
      color: colors.apple,
    },
    description: {
    },
    validation: {
      color: colors.apple,
    },
  },
  [validations.WAIT]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.selectiveYellow,
      borderWidth: 2,
    },
    title: {
      color: colors.selectiveYellow,
    },
    description: {
    },
    validation: {
      color: colors.selectiveYellow,
    },
  },
  [validations.BAD]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.bittersweet,
      borderWidth: 2,
    },
    title: {
      color: colors.bittersweet,
    },
    description: {
    },
    validation: {
      color: colors.bittersweet,
    },
  },
}

export default CustomDatePicker
