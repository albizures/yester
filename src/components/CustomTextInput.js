import React from 'react'
import PropTypes from 'prop-types'
import { View, ViewPropTypes, StyleSheet, TextInput, Image } from 'react-native'
import colors from '../utils/colors'
import Translate from '../components/Translate'
import icons from '../utils/icons'

const states = {
  ENABLED: 'ENABLED',
  DISABLED: 'DISABLED',
}

export const validations = {
  NONE: 'NONE',
  GOOD: 'GOOD',
  WAIT: 'WAIT',
  BAD: 'BAD',
}

const CustomTextInput = (props) => {
  const {title, editable, description, value, validation,
    validationMessage} = props
  const stateStyle = stateStyles[editable ? states.ENABLED : states.DISABLED]
  const validationStyle = validationStyles[validation]

  var descriptionLabel
  if (description != null) {
    descriptionLabel = <Translate keyName={description}
      style={styles.description} />
  }

  var validationLabel
  if (validation !== validations.NONE) {
    validationLabel = <Translate keyName={validationMessage}
      style={[styles.validation, validationStyle.validation]} />
  }

  return (
    <View>
      <View style={[styles.labelContainer]}>
        <Translate keyName={title}
          style={[styles.tile, stateStyle.title, validationStyle.title]} />
      </View>
      <View style={[styles.input, stateStyle.input, validationStyle.input]}>
        <TextInput {...props}
          value={value}
          paddingHorizontal={20}
          height={50}
          width={250}
          color={(editable) ? colors.mineShaft : colors.white}
        />
        <Image source={icons.fb} style={{ width: 20, height: 20, marginRight: 9 }} />
      </View>
      <View style={[styles.labelContainer, {justifyContent: 'flex-end'}]}>
        {validationLabel}
      </View>
      <View style={[styles.labelContainer, {justifyContent: 'flex-end'}]}>
        {descriptionLabel}
      </View>
    </View>
  )
}

CustomTextInput.propTypes = {
  style: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
  editable: PropTypes.bool,
  description: PropTypes.string,
  value: PropTypes.string,
  validation: PropTypes.oneOf(Object.keys(validations)),
  validationMessage: PropTypes.string,
}

CustomTextInput.defaultProps = {
  editable: true,
  validation: validations.NONE,
}

const styles = StyleSheet.create({
  input: {
    flex: 1,
    flexDirection: 'row',
    width: 300,
    maxHeight: 50,
    borderRadius: 25,
    shadowColor: colors.mischka,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowRadius: 15,
    shadowOpacity: 1.0,
    elevation: 10,
  },
  title: {
    position: 'absolute',
    left: 0,
    top: 0,
    fontSize: 12,
  },
  description: {
    fontSize: 12,
    color: colors.mineShaft,
  },
  labelContainer: {minHeight: 24},
  validation: {
  },
})

const stateStyles = {
  [states.ENABLED]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.white,
      borderWidth: 0.5,
    },
    title: {
      color: colors.governorBay,
    },
    description: {
    },
    validation: {
    },
  },
  [states.DISABLED]: {
    input: {
      backgroundColor: colors.mischka,
      borderColor: colors.mischka,
      borderWidth: 0.5,
    },
    title: {
      color: colors.mischka,
    },
    description: {
    },
    validation: {
    },
  },
}

const validationStyles = {
  [validations.NONE]: {
    input: {
    },
    title: {
    },
    description: {
    },
    validation: {
    },
  },
  [validations.GOOD]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.apple,
      borderWidth: 2,
    },
    title: {
      color: colors.apple,
    },
    description: {
    },
    validation: {
      color: colors.apple,
    },
  },
  [validations.WAIT]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.selectiveYellow,
      borderWidth: 2,
    },
    title: {
      color: colors.selectiveYellow,
    },
    description: {
    },
    validation: {
      color: colors.selectiveYellow,
    },
  },
  [validations.BAD]: {
    input: {
      backgroundColor: colors.white,
      borderColor: colors.bittersweet,
      borderWidth: 2,
    },
    title: {
      color: colors.bittersweet,
    },
    description: {
    },
    validation: {
      color: colors.bittersweet,
    },
  },
}

export default CustomTextInput
