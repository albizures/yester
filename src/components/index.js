import React from 'react'
import suppplant from 'suppplant'
import { Text, StyleSheet } from 'react-native'
import Translate from './Translate'
import colors from '../utils/colors'

const createStyledText = (defaultStyles) => ({ keyName, data, text, style, ...props }) => {
  if (text !== undefined) {
    return (
      <Text {...props}
        style={defaultStyles.concat(style)} >
        {suppplant(text, data)}
      </Text>
    )
  }

  if (keyName !== undefined) {
    return (
      <Translate {...props}
        style={defaultStyles.concat(style)}
        keyName={keyName}
        data={data} />
    )
  }

  return null
}

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Karla-Regular',
    color: colors.mineShaft,
    textAlign: 'left',
  },
  heading1: {
    fontSize: 50,
    fontWeight: 'bold',
  },
  heading2: {
    fontSize: 25,
    fontWeight: 'bold',
  },
  heading3: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  heading4: {
    fontSize: 18,
  },
  heading5: {
    fontSize: 16,
  },
  title: {
    fontSize: 20,
  },
  body1: {
    fontSize: 14,
    fontWeight: 'bold',
  },
  body2: {
    fontSize: 14,
  },
  description: {
    fontSize: 14,
  },
})

export const Heading1 = createStyledText([styles.text, styles.heading1])
export const Heading2 = createStyledText([styles.text, styles.heading2])
export const Heading3 = createStyledText([styles.text, styles.heading3])
export const Heading4 = createStyledText([styles.text, styles.heading4])
export const Heading5 = createStyledText([styles.text, styles.heading5])
export const Title = createStyledText([styles.text, styles.title])
export const Body1 = createStyledText([styles.text, styles.body1])
export const Body2 = createStyledText([styles.text, styles.body2])
export const Description = createStyledText([styles.text, styles.description])
