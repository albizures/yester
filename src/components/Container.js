import React from 'react'
import PropTypes from 'prop-types'
import { View, Platform, ScrollView, RefreshControl, StyleSheet, ViewPropTypes } from 'react-native'

import ConditionalWrapper from './ConditionalWrapper'
import Loading from './Loading'
import colors from '../utils/colors'

const top = Platform.OS === 'ios' ? 20 : 0

const Container = (props) => {
  const { scroll, isLoading, children, topBar, onRefresh, refreshing, style, scrollRef } = props
  const refreshControl = (scroll && onRefresh && refreshing) ? (
    <RefreshControl
      refreshing={refreshing}
      onRefresh={onRefresh}
    />
  ) : undefined

  const ScrollViewProps = {
    refreshControl,
    style: styles.flex,
    ref: scrollRef,
  }

  return (
    <View style={[styles.container, style, !topBar ? { paddingTop: top } : undefined]}>
      <Loading top={top} isLoading={isLoading}>
        {topBar}
        <ConditionalWrapper component={ScrollView} condition={scroll} props={ScrollViewProps}>
          {children}
        </ConditionalWrapper>
      </Loading>
    </View>
  )
}

Container.propTypes = {
  onRefresh: PropTypes.func,
  refreshing: PropTypes.bool,
  children: PropTypes.node.isRequired,
  topBar: PropTypes.node,
  isLoading: PropTypes.bool,
  scroll: PropTypes.bool,
  style: ViewPropTypes.style,
  scrollRef: PropTypes.object,
}

const styles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: colors.athensGray,
  },
})

export default Container
