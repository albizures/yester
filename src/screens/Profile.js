import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Image, Dimensions, StyleSheet } from 'react-native'
import { Heading2, Description, Heading5 } from '../components'
import Container from '../components/Container'
import TopBar from '../components/TopBar'
import Divider from '../components/Divider'
import colors from '../utils/colors'
import icons from '../utils/icons'
import withUser, { shapeContextUser } from '../components/withUser'
import { isSetupFinished } from '../utils/session'
import { screen } from '../utils/analytics'

class Profile extends Component {
  static propTypes = {
    contextUser: PropTypes.shape(shapeContextUser).isRequired,
    navigation: PropTypes.object.isRequired,
  }

  constructor (props) {
    super()
    const {
      contextUser: { user },
    } = props
    this.state = { user }
  }

  willFocusListener = null

  componentDidMount () {
    const { navigation } = this.props
    const { addListener } = navigation
    this.willFocusListener = addListener('willFocus', this.load)
  }

  componentWillUnmount () {
    this.willFocusListener.remove()
  }

  load = () => {
    screen('Profile', {})
  }

  onPressEdit = async () => {
    const { navigation } = this.props
    const { user } = this.state
    const { params } = await isSetupFinished(user)
    return navigation.navigate('SetupBirthDate', params)
  }

  onPressVerify = async () => {
    const { navigation } = this.props
    const { email, emailVerified } = this.state.user
    if (!emailVerified) {
      return navigation.navigate('ConfirmAccount', { email: email, signUpVerify: false })
    }
  }

  render () {
    const { name, email, country, state, gender, birthPlace, emailVerified } = this.state.user
    const location =
      birthPlace !== undefined ? birthPlace : `${country}, ${state ? state.substring(3, 5) : ''}`

    const topBar = <TopBar title='profile.title' />

    let verificationElement = (
      <Description
        keyName='profile.verifiedEmail.confirmed'
        style={{ color: 'green', marginTop: height * 0.01 }}
      />
    )
    if (!emailVerified) {
      verificationElement = (
        <View style={styles.verification}>
          <Description keyName='profile.verifiedEmail.unconfirmed' style={{ color: 'red' }} />
          <Heading5
            keyName='profile.verifiedEmail.confirm'
            style={styles.confirmText}
            onPress={this.onPressVerify}
          />
        </View>
      )
    }

    return (
      <Container topBar={topBar} style={styles.container}>
        <View style={styles.topFlex}>
          <Image
            source={gender === 'male' ? icons.profileMan : icons.profileWoman}
            style={styles.image}
          />
          <Heading2 text={name} style={styles.nameText} />
          {
            // TODO get the number of answered stories.
            // <Heading5 text='3 stories' style={{marginBottom: 30}} />
          }
        </View>

        <View style={styles.bottomFlex}>
          <Divider style={{ width: 323 }} />
          <View style={styles.item}>
            <Description keyName='profile.email' />
            <Heading5 text={email} style={{ fontWeight: 'bold' }} onPress={this.onPressVerify} />
            {verificationElement}
          </View>
          <Divider style={{ width: 323 }} />
          <View style={styles.item}>
            <Description keyName='profile.location' />
            <Heading5 text={location} style={{ fontWeight: 'bold' }} />
          </View>
          <Divider style={{ width: 323 }} />
          <View style={styles.edit}>
            <Description
              keyName='profile.edit'
              style={styles.editLabel}
              onPress={this.onPressEdit}
            />
          </View>
        </View>
      </Container>
    )
  }
}

const { width, height } = Dimensions.get('window')
const styles = StyleSheet.create({
  container: {
    width,
    height,
  },
  topFlex: {
    flex: 0.35,
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingTop: height * 0.06,
  },
  bottomFlex: {
    flex: 0.65,
    alignItems: 'center',
    paddingHorizontal: width * 0.07,
  },
  nameText: {
    marginTop: 17,
    color: colors.governorBay,
  },
  image: {
    width: 100,
    height: 100,
  },
  item: {
    width: 300,
    minHeight: height * 0.1,
    alignItems: 'flex-start',
    paddingVertical: height * 0.025,
  },
  verification: {
    flexDirection: 'row',
    width: 300,
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    marginTop: height * 0.015,
  },
  confirmText: {
    fontWeight: 'bold',
    color: colors.governorBay,
  },
  edit: {
    width: 300,
    height: 80,
    alignItems: 'center',
    paddingTop: 40,
  },
  editLabel: {
    fontWeight: 'bold',
    textDecorationLine: 'underline',
  },
})

export default withUser(Profile)
