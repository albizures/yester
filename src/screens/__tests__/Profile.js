import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Container } from '../../components/Container'

import Profile from '../Profile'

const navigation = { navigate: jest.fn() }

describe('src/screens/Profile.js', () => {
  it('should render', () => {
    const tree = renderer.create(<Profile navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()

    const shallowHome = shallow(<Profile navigation={navigation} />)
    expect(shallowHome.type).toBe(Container)
  })
})
