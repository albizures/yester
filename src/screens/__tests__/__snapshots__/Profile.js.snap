// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/Profile.js should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F6F7F9",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1E1437",
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 70,
        "paddingTop": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      My Profile
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#313131",
          "fontFamily": "Karla-Regular",
          "textAlign": "left",
        },
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
        },
        Object {
          "color": "#4241B3",
          "marginTop": 153,
        },
      ]
    }
  >
    Luis Galvez
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#313131",
          "fontFamily": "Karla-Regular",
          "textAlign": "left",
        },
        Object {
          "fontSize": 16,
        },
        Object {
          "marginBottom": 30,
        },
      ]
    }
  >
    3 stories
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 26,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#D5D3DD",
            "borderColor": "#D5D3DD",
            "borderWidth": 0,
            "height": 1,
            "marginHorizontal": 22,
          },
          Object {
            "marginLeft": 0,
            "width": 323,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "textAlign": "left",
            },
            Object {
              "fontSize": 14,
            },
            undefined,
          ]
        }
      >
        Email
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "textAlign": "left",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        luis@yester.app
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#D5D3DD",
            "borderColor": "#D5D3DD",
            "borderWidth": 0,
            "height": 1,
            "marginHorizontal": 22,
          },
          Object {
            "marginLeft": 0,
            "width": 323,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "textAlign": "left",
            },
            Object {
              "fontSize": 14,
            },
            undefined,
          ]
        }
      >
        Location
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "textAlign": "left",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Guatemala, Guatemala
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#D5D3DD",
            "borderColor": "#D5D3DD",
            "borderWidth": 0,
            "height": 1,
            "marginHorizontal": 22,
          },
          Object {
            "marginLeft": 0,
            "width": 323,
          },
        ]
      }
    />
  </View>
</View>
`;
